gdal/alg/gdalgrid.cpp

```diff
@@ -2155,7 +2155,8 @@ void GDALGridContextFree(GDALGridContext* psContext)
         CPLFree(psContext->pabyZ);
         if( psContext->sExtraParameters.psTriangulation )
             GDALTriangulationFree( psContext->sExtraParameters.psTriangulation );
-        delete psContext->poWorkerThreadPool;
+        if (psContext->poWorkerThreadPool)
```

Comment:
I'm not sure really. I've just learnt that `delete NULL` is ok. I was chasing the segfault when running the integration tests of these [haskell bindings](https://github.com/albertov/bindings-gdal) which are multi-threaded so perhaps the segfault was caused by the un-initialized contour generator in another thread. I'll run them again without this change and report back